### PR TITLE
CTCTRADERS-1832  update regex to disallow invalid characters

### PR DESCRIPTION
**Pull request by kenneth-rayner**

### CODE REVIEW

File: test/forms/AddSecurityDetailsFormProviderSpec.scala [hunk 1]
```diff
@@ -21,10 +21,9 @@ import play.api.data.FormError
 
 class AddSecurityDetailsFormProviderSpec extends BooleanFieldBehaviours {
 
-  val requiredKey = "addSecurityDetails.error.required"
-  val invalidKey  = "error.boolean"
-
-  val form = new AddSecurityDetailsFormProvider()()
+  private val requiredKey = "addSecurityDetails.error.required"
```
[the remark this answers is not in the document]
Comment (kenneth-rayner) [reply test/forms/AddSecurityDetailsFormProviderSpec.scala:24]: I did reformat some of the code using intelliJ's control A, control Alt L

File: app/forms/AccessCodeFormProvider.scala [hunk 2]
```diff
@@ -18,18 +18,19 @@ package forms
...
   val accessCodeLength = 4
-  val accessCodeRegex  = "^[0-9A-Za-z]{4}$"
 
   def apply(): Form[String] =
     Form(
       "value" -> text("accessCode.error.required")
         .verifying(
-          StopOnFirstFail[String](exactLength(accessCodeLength, "accessCode.error.length"), regexp(accessCodeRegex, "accessCode.error.invalidCharacters")))
+          StopOnFirstFail[String](exactLength(accessCodeLength, "accessCode.error.length"),
+                                  regexp(alphaNumericRegex, "accessCode.error.invalidCharacters", Seq.empty)))
```
[the remark this answers is not in the document]
Comment (kenneth-rayner) [reply app/forms/AccessCodeFormProvider.scala:34]: Did second option

File: test/forms/ConsignorNameFormProviderSpec.scala [hunk 3]
```diff
@@ -49,5 +51,17 @@ class ConsignorNameFormProviderSpec extends StringFieldBehaviours {
...
     )
+
+    "must not bind strings with invalid characters" in {
+
+      val expectedError          = FormError(fieldName, invalidKey)
+      val generator: Gen[String] = RegexpGen.from(s"[!£^*(){}_+=<>:;|`~,±üçñèé@]{35}")
+
+      forAll(generator) {
+        invalidString =>
+          val result: Field = form.bind(Map(fieldName -> invalidString)).apply(fieldName)
```
[the remark this answers is not in the document]
Comment (kenneth-rayner) [reply test/forms/ConsignorNameFormProviderSpec.scala:62]: possible tech debt?

**Comment by snehaprathipat1** [comment]
test please

**Pull request merged**